internal/util/url.go

```diff
@@ -45,18 +48,50 @@ const (
 
 // FetchResource fetches a resource given a URL. The supported schemes are http, data, and oem.
 func FetchResource(l *log.Logger, u url.URL) ([]byte, error) {
+	var data []byte
+
+	dataReader, err := FetchResourceAsReader(l, u)
+	if err != nil {
+		return nil, err
+	}
+
+	defer dataReader.Close()
+
+	data, err = ioutil.ReadAll(dataReader)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+// Calls umount() after close
+type UnmountCloser struct {
+	umount func()
+	io.Closer
+}
+
+func (f UnmountCloser) Close() error {
```

Comment:
If you stored the logger instead of a function usage would be a little simpler to follow and close would be:

```go
func (f UnmountCloser) Close() error {
    defer umountOEM(f.logger)
    return f.Closer.Close()
}
```

Or perhaps call it a ReadUnmounter so you implement ReadCloser and avoid wrapping the underlying file in yet another struct below.